OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
```

Comment:
Does it make sense to close off all derived classes of Geometry from extending `generateDecorations()`? I like that the task is simplified for derived Geometry, but is it reasonable to close them off completely from having a say in how they draw themselves based on the whether *fixed* decorations is requested or making use of display *hints*? 

One can imagine a `LinePathGeometry` as a functional and drawing replacement for GeometryPath, in which case we would want the geometry to draw differently for many of the `ModelDisplayHints` options.